solidity/contracts/bridge/Bridge.sol

```diff
@@ -1918,28 +1923,81 @@ contract Bridge is Ownable, EcdsaWalletOwner {
+    ///         longer be marked as the active wallet (if it was marked as such).
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @param redeemerOutputScript  The redeemer's length-prefixed output
+    ///        script (P2PKH, P2WPKH, P2SH or P2WSH)
+    /// @dev Requirements:
+    ///      - The redemption request identified by `walletPubKeyHash` and
+    ///        `redeemerOutputScript` must exist
+    ///      - The amount of time defined by `redemptionTimeout` must have
+    ///        passed since the redemption was requested (the request must be
+    ///        timed-out).
+    function notifyRedemptionTimeout(
+        bytes20 walletPubKeyHash,
+        bytes calldata redeemerOutputScript
+    ) external {
+        uint256 redemptionKey = uint256(
+            keccak256(abi.encodePacked(walletPubKeyHash, redeemerOutputScript))
+        );
+        RedemptionRequest storage request = pendingRedemptions[redemptionKey];
+
+        require(request.requestedAt > 0, "Redemption request does not exist");
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            request.requestedAt + redemptionTimeout < block.timestamp,
+            "Redemption request has not timed out"
+        );
+
+        // Update the wallet's pending redemptions value
+        Wallets.Wallet storage wallet = wallets.registeredWallets[
+            walletPubKeyHash
+        ];
+        wallet.pendingRedemptionsValue -=
+            request.requestedAmount -
+            request.treasuryFee;
+
+        // Return the requested amount of tokens to the redeemer
+        bank.transferBalance(request.redeemer, request.requestedAmount);
+
+        // Notice that there is no need to check if `timedOutRedemptions`
+        // mapping already contains the redemption key because
+        // `requestRedemption` blocks requests targeting non-live wallets.
+        // Because `notifyRedemptionTimeout` changes the state of any live-wallet
+        // after the first call, there is no possibility that the given
+        // redemption key could be reported as timed-out multiple times.
+        // At the same time, if the given redemption key was already marked as
+        // fraudulent due to an amount-related fraud, it will not be possible
+        // to report a timeout on it since it will not be present in
+        // `pendingRedemptions` mapping.
+
+        // Move the redemption from pending redemptions to timed-out redemptions
+        timedOutRedemptions[redemptionKey] = request;
+        delete pendingRedemptions[redemptionKey];
+
+        if (
+            wallet.state == Wallets.WalletState.Live ||
+            wallet.state == Wallets.WalletState.MovingFunds
+        ) {
+            // Propagate timeout consequences to the wallet
+            wallets.notifyRedemptionTimedOut(walletPubKeyHash);
+        }
+        // TODO: should the notifier be reimbursed?
```

Comment:
No, according to https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-6.adoc#notify-redemption-timeout. Notifier should be rewarded some % of slashed tokens.

Reply:
Done in 0e9b80110b1479820dfc3cbefcdf5054bf36b3df.